test/extended/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -0,0 +1,318 @@
+				o.Expect(err).To(o.HaveOccurred())
+			}
+
+			g.By("wait for PDB to become stable")
+			expectedStatus = policyv1.PodDisruptionBudgetStatus{
+				DisruptionsAllowed: 1,
+				CurrentHealthy:     3,
+				DesiredHealthy:     2,
+				ExpectedPods:       3,
+			}
+			pdb, err = waitForPDBStatus(oc.KubeClient().PolicyV1(), metav1.ObjectMeta{Name: ifHealthyBudgetPDBName, Namespace: oc.Namespace()}, e2e.PodStartTimeout+podBecomesReadyTimeout, expectedStatus)
+			if err != nil {
+				g.Fail(fmt.Sprintf("error occurred while waiting for PDB status: %v\nexpected %#v, got %#v", err, expectedStatus, pdbStatus(pdb)))
+			}
+
+			g.By("check pods are ready")
+			pods, err = exutil.WaitForPods(oc.KubeClient().CoreV1().Pods(oc.Namespace()), podsLabelSelector, func(pod v1.Pod) bool {
+				return exutil.CheckPodIsReady(pod) && pod.DeletionTimestamp == nil
+			}, 3, e2e.PodStartTimeout+podBecomesReadyTimeout)
+
+			shouldTerminatePods := sets.New[string]()
+
+			g.By("only one disruption of ready pods allowed")
+			firstToEvictPod := pods[0]
+			eviction := newV1Eviction(oc.Namespace(), firstToEvictPod, metav1.DeleteOptions{})
+			err = oc.KubeClient().PolicyV1().Evictions(oc.Namespace()).Evict(context.Background(), eviction)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			shouldTerminatePods.Insert(firstToEvictPod)
+
+			g.By("second eviction should not succeed")
+			secondToEvictPod := pods[1]
+			eviction = newV1Eviction(oc.Namespace(), secondToEvictPod, metav1.DeleteOptions{})
+			err = oc.KubeClient().PolicyV1().Evictions(oc.Namespace()).Evict(context.Background(), eviction)
+			o.Expect(err).To(o.HaveOccurred())
+
+			g.By("wait for the new pod to be running (running but not ready)")
+			pods, err = exutil.WaitForPods(oc.KubeClient().CoreV1().Pods(oc.Namespace()), podsLabelSelector, func(pod v1.Pod) bool {
+				return exutil.CheckPodIsRunning(pod) && !exutil.CheckPodIsReady(pod) && pod.DeletionTimestamp == nil
+			}, 1, e2e.PodStartTimeout)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("wait for PDB to update CurrentHealthy == 2 and DisruptionsAllowed == 0")
+			expectedStatus = policyv1.PodDisruptionBudgetStatus{
+				DisruptionsAllowed: 0,
+				CurrentHealthy:     2,
+				DesiredHealthy:     2,
+				ExpectedPods:       3,
+			}
+			pdb, err = waitForPDBStatus(oc.KubeClient().PolicyV1(), metav1.ObjectMeta{Name: ifHealthyBudgetPDBName, Namespace: oc.Namespace()}, e2e.PodStartTimeout, expectedStatus)
+			if err != nil {
+				g.Fail(fmt.Sprintf("error occurred while waiting for PDB status: %v\nexpected %#v, got %#v", err, expectedStatus, pdbStatus(pdb)))
+			}
+
+			g.By("eviction of the new running but not ready pod should succeed")
+			thirdNewToEvictPod := pods[0]
+			eviction = newV1Eviction(oc.Namespace(), thirdNewToEvictPod, metav1.DeleteOptions{})
+			err = oc.KubeClient().PolicyV1().Evictions(oc.Namespace()).Evict(context.Background(), eviction)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			shouldTerminatePods.Insert(thirdNewToEvictPod)
```

Comment:
will the evicted pod here come back eventually ? This is a little confusing though

Reply:
the evicted pod will be deleted, that is what we check with shouldTerminatePods